### PR TITLE
Avoid closing cursor when consumers are disconnected

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

While investigating the failures of `SequenceIdWithErrorTest` which are also related to #2148, I've identified these 2 issues: 

 * Getting a "fenced" error during a consumer subscribe operation doesn't trigger the closing of the topic, thus the recovery might not be triggered if there are not producers
 * When closing a topic, we usually "ignore" managed ledger failures (in the sense that we rely on recovery later on). The only place where we don't ignore the errors on is on the `Subscription.close()` method. This method should only deal with disconnecting existing consumers and it should not close the ManagedLedger cursor. The cursor is closed when the managed ledger is closed with the topic.

### Modifications
 * Do not close cursor when the subscription is closed, cursor will be close when managed ledger is closed
 * When we do get a "fenced" error during a subscribe operation, we should close the topic, in same way as we do for add entry operations failed.
 * Re-enabled `SequenceIdWithErrorTest`

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/SequenceIdWithErrorTest.java [hunk 1]
```diff
@@ -39,7 +39,7 @@
     /**
      * Test that sequence id from a producer is correct when there are send errors
      */
-    @Test(enabled = false)
+    @Test
```
Comment (rdhabalia) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/SequenceIdWithErrorTest.java]: did you try with `invocationCount=20`? I think earlier when I spent sometime to debug this issue,  I tried to close-topic on subscription-fenced ledger exception but still test was failing intermittently and I realized topic was stuck at [schemaVersionFuture](https://github.com/apache/incubator-pulsar/blob/master/pulsar-broker/src/main/java/org/apache/pulsar/broker/service/ServerCnx.java#L832) so, produce was timing out.

Comment (merlimat) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/impl/SequenceIdWithErrorTest.java]: Yes, I tried with 300 invocation. This pr is just part of the fix, the actual fix for schema future is at #2148

Comment (rdhabalia) [reply pulsar-broker/src/test/java/org/apache/pulsar/client/impl/SequenceIdWithErrorTest.java]: I see, I didn't see it. make sense.

**Comment by rdhabalia** [comment]
retest this please

**Pull request merged**